doc/design/web.md

```diff
@@ -0,0 +1,153 @@
+# Web Interface design
+
+This design doc will talk about features and web pages needed to let users manage cloud paddle jobs.
+
+## Feature List
+
+- Account Management
+    - Registration, send email to inform if registration succeeded
+    - Account Login/Logout
+    - Password changing, find back
+    - Download SSL keys
+- Jupiter Notebook
+    - Private Jupiter Notebook environment to run Python scripts
+    - Private workspace
+    - Submit job from Jupiter Notebook
+- Job Dashboard
+    - Job history and currently running jobs
+    - Performance Monitoring
+    - Quota Monitoring
+- Datasets
+    - Public Dataset viewing
+    - Upload/Download private datasets
+    - Share datasets
+- Models
+    - Upload/Download models file
+    - Share/Publish Models
+- Paddle Board
+    - Training metrics visualization
+        - cost
+        - evaluator
+        - user-defined metrics
+- Serving
+    - Submit serving instances
+    - Deactivate serving
+    - Serving performance monitoring
+
+## Account Management
+
+Account management page is designed to satisfy multi-tenant use cases. One account should have a unique account ID for login, and this account owns one access key to one unique [Kubernetes namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/) cluster. Multiple users can log in to this account ID and operate jobs and data files. The only "master user" can do modifications like increase quota or manage account settings.
+
+One example is [AWS IAM](https://aws.amazon.com/iam/?nc2=h_m1), but we can do more simpler than that.
+
+The current implementation under this repo can only have one user for one Kubernetes namespace. We can implement multi-tenant in the near future.
+
+Once a user logged in, s/he will be redirected to the "Job Dashboard" page.
+
+## Jupiter Notebook
+
+Start a [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) using image `docker.paddlepaddle.org/book` in the Kubernetes cluster and add an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) endpoint when a user first enters the notebook page. This is already implemented.
+
+<img src="pictures/notebook.png" width="500px" align="center">
+
+Users can write a program in python in the web page and save their programs, which will be saved at cloud storage. Users also can run a script like below to submit a cluster training job:
+
+```python
+create_cloud_job(
+    name,
+    num_trainer,
+    mem_per_trainer,
+    gpu_per_trainer,
+    cpu_per_trainer,
+    num_ps,
+    mem_per_ps,
+    cpu_per_ps,
+)
+```
+
+After this, there will be a job description and performance monitoring pages able to view at "Job Dashboard"
+
+## Job Dashboard
+
+### Job history and currently running jobs
+
+A web page containing a table to list jobs satisfying user's filters. The user can only list jobs that were submitted by themselves.
+
+| jobname | start time | age      | success | fails | actions |
```

Comment:
Maybe we need mor information on the Job list, such as `PS_READY, PS_TOTAL, TRAINER_READY, TRAINER_TOTAL`.